drivers/accel/adxl367/adxl367.c

```diff
@@ -0,0 +1,1310 @@
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Configures the FIFO feature. Uses ADXL367_14B_CHID read mode as
+ * 		default.
+ *
+ * @param dev          - The device structure.
+ * @param mode         - FIFO mode selection.
+ *                       Example: ADXL367_FIFO_DISABLED,
+ *								  ADXL367_OLDEST_SAVED,
+ *								  ADXL367_STREAM_MODE,
+ *								  ADXL367_TRIGGERED_MODE
+ * @param format	   - FIFO format selection.
+ * 				         Example:  ADXL367_FIFO_FORMAT_XYZ,
+ *								   ADXL367_FIFO_FORMAT_X,
+ *								   ADXL367_FIFO_FORMAT_Y,
+ *								   ADXL367_FIFO_FORMAT_Z,
+ *								   ADXL367_FIFO_FORMAT_XYZT,
+ *	 							   ADXL367_FIFO_FORMAT_XT,
+ *								   ADXL367_FIFO_FORMAT_YT,
+ *								   ADXL367_FIFO_FORMAT_ZT,
+ *								   ADXL367_FIFO_FORMAT_XYZA,
+ *								   ADXL367_FIFO_FORMAT_XA,
+ *								   ADXL367_FIFO_FORMAT_YA,
+ *								   ADXL367_FIFO_FORMAT_ZA
+ * @param sets_nb 	   - Specifies the number of samples sets to store in the FIFO.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_fifo_setup(struct adxl367_dev *dev,
+			   enum adxl367_fifo_mode mode,
+			   enum adxl367_fifo_format format,
+			   uint8_t sets_nb)
+{
+	int32_t ret;
+
+	ret = adxl367_set_fifo_mode(dev, mode);
+	if (ret)
+		return ret;
+
+	ret = adxl367_set_fifo_format(dev, format);
+	if (ret)
+		return ret;
+
+	ret = adxl367_set_fifo_sample_sets_nb(dev, sets_nb);
+	if (ret)
+		return ret;
+
+	return adxl367_set_fifo_read_mode(dev, ADXL367_14B_CHID);
+}
+
+/***************************************************************************//**
+ * @brief Reads raw values from FIFO. If, after setting FIFO mode, any of x, y, z,
+ * 		temp or adc aren't selected, assign NULL pointer. Uses ADXL367_14B_CHID
+ * 		read mode as default.
+ *
+ * @param dev       - The device structure.
+ * @param x  	    - X axis raw data buffer. If not used, assign NULL.
+ * @param y  	    - Y axis raw data buffer. If not used, assign NULL.
+ * @param z  	    - Z axis raw data buffer. If not used, assign NULL.
+ * @param temp_adc  - Data buffer for temperature or adc raw data. If not
+ * 				used assign NULL.
+ * @param sets_nb   - Number of sample sets to be read.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_read_raw_fifo(struct adxl367_dev *dev, int16_t *x, int16_t *y,
+			      int16_t *z, int16_t *temp_adc, uint16_t sets_nb)
+{
+	int32_t ret;
+	uint16_t entries_nb, i, stored_entr = 0;
+	entries_nb = sets_nb * samples_per_set;
+
+	// allocate buffer for local storage, each entry has 2 bytes
+	uint8_t buffer[entries_nb * 2], id;
+	memset(buffer, 0, entries_nb * 2);
+
+	//wait for FIFO to have at least sets_nb * samples_per_set entries
+	while (stored_entr < entries_nb) {
```

Comment:
I would check sets_nb value before. If the value is invalid, it will enter an infinite loop here.